src/components/SourcesTree.js

```diff
@@ -119,6 +121,31 @@ let SourcesTree = React.createClass({
     return Svg("folder");
   },
 
+  onContextMenu(event, item) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    if (!isDirectory(item)) {
+      const source = item.contents.get("url");
+      const copySourceUrl = {
+        id: "node-menu-copy-source",
+        label: "Copy Source URL",
+        accesskey: "X",
+        disabled: false,
+        click: () => copyToTheClipboard(source)
+      };
+
+      let items = [];
+
+      if (isEnabled("copySource")) {
+        items.push(copySourceUrl);
+      }
+
+      showMenu(event, items);
```

Comment:
hmm, lets move `let items` and `showMenu(event, items);` out of the the condition